src/pyscipopt/scip.pxi

```diff
@@ -5553,10 +5576,371 @@ cdef class Model:
+        cdef SCIP_Bool downvalid
+        cdef SCIP_Bool upvalid
+        cdef SCIP_Bool downinf
+        cdef SCIP_Bool upinf
+        cdef SCIP_Bool downconflict
+        cdef SCIP_Bool upconflict
+        cdef SCIP_Bool lperror
+
+        if integral:
+            PY_SCIP_CALL(SCIPgetVarStrongbranchInt(self._scip, var.scip_var, itlim, idempotent, &down, &up, &downvalid, 
+                                                   &upvalid, &downinf, &upinf, &downconflict, &upconflict, &lperror))
+        else:
+            PY_SCIP_CALL(SCIPgetVarStrongbranchFrac(self._scip, var.scip_var, itlim, idempotent, &down, &up, &downvalid, 
+                                                    &upvalid, &downinf, &upinf, &downconflict, &upconflict, &lperror))
+
+        return down, up, downvalid, upvalid, downinf, upinf, downconflict, upconflict, lperror
+
+    def updateVarPseudoCost(self, Variable var, valdelta, objdelta, weight):
+        """Updates the pseudo costs of the given variable and the global pseudo costs after a change of valdelta
+         in the variable's solution value and resulting change of objdelta in the LP's objective value.
+         Update is ignored if objdelts is infinite. Weight is in range (0, 1], and affects how it updates
+         the global weighted sum.
+
+         :param Variable var: Variable whos pseudo cost will be updated
+         :param valdelta: The change in variable value (e.g. the fractional amount removed or added by branching)
+         :param objdelta: The change in objective value of the LP after valdelta change of the variable
+         :param weight: the weight in range (0,1] of how the update affects the stored weighted sum.
+         """
+
+        PY_SCIP_CALL(SCIPupdateVarPseudocost(self._scip, var.scip_var, valdelta, objdelta, weight))
+
+    def getBranchScoreMultiple(self, Variable var, gains):
+        """Calculates the branching score out of the gain predictions for a branching with
+        arbitrary many children.
+
+        :param Variable var: variable to calculate the score for
+        :param gains: list of gains for each child.
+        """
+
+        assert isinstance(gains, list)
+        nchildren = len(gains)
+
+        cdef int _nchildren = nchildren
+        _gains = <SCIP_Real*> malloc(_nchildren * sizeof(SCIP_Real))
+        for i in range(_nchildren):
+            _gains[i] = gains[i]
+
+        score = SCIPgetBranchScoreMultiple(self._scip, var.scip_var, _nchildren, _gains)
+
+        free(_gains)
+
+        return score
+
     def getTreesizeEstimation(self):
         """Get an estimation of the final tree size """
         return SCIPgetTreesizeEstimation(self._scip)
 
+
+    def getBipartiteGraphRepresentation(self, prev_col_features=None, prev_edge_features=None, prev_row_features=None,
+                                        static_only=False, suppress_warnings=False):
+        """This function generated the bipartite graph representation of an LP, which was first used in
```

Comment:
generates instead of generated. Or if you want to say that it generated the representation first used in the paper, I think you need to say something like "... of an LP used in the following paper". Not a grammar expert though, so might be wrong.